src/services/telemetry.ts

```diff
@@ -84,3 +84,16 @@ function getTelemetrySettings(): number {
       return 3;
   }
 }
+
+let service: TelemetryService;
+
+export let createTelemetryService = (mixpanelToken: string) => {
+  if (!service) {
+    service = new TelemetryService(mixpanelToken);
+  }
+  return service;
+}
+
+export let getTelemetryService = () => {
+  return service;
+}
```

Comment:
Again, confused by the necessity of this. Only one telem service should ever exist in extension.ts if we use a constructor, why do we need this check?

Reply:
If other services need it, we'd need to pass it in the constructor. This way, instead of passing it to all the services that need it, we can just call getTelemetryService instead, which imo is better functionality wise. EG: say a service that is a child of another service, which is a child of another service, which is defined in the extension needs it, we'd need to add it to the constructor of all the classes in between. This way we can just call a function and be done with it

Reply:
Gotcha. Again, naming issue at this point. We'll keep it as a singleton but rename along with everything else that is refactored.